scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -289,9 +289,16 @@ def handle_modules(self):
 
 
     def report(self):
-        if self.options.list_test_duplicates:
-            self.report_duplicates()
-            return 0
+        if self.options.list_test_duplicates or self.options.test_duplicates_as_error:
```

Comment:
instead of using the report workflow, it would make sense to do that during discovery, i.e.:

```
diff --git a/scripts/pylib/twister/twisterlib/testplan.py b/scripts/pylib/twister/twisterlib/testplan.py
index ad2486e4b0..818ca30b25 100755
--- a/scripts/pylib/twister/twisterlib/testplan.py
+++ b/scripts/pylib/twister/twisterlib/testplan.py
@@ -175,6 +175,11 @@ class TestPlan:
         for _, ts in self.testsuites.items():
             self.scenarios.append(ts.id)

+        seen = set()
+        dupes = [x for x in self.scenarios if x in seen or seen.add(x)]
+        if dupes:
+            raise TwisterRuntimeError(f"Duplicate test scenarios found: {dupes}.")
+
         self.parse_configuration(config_file=self.env.test_config)
         self.add_configurations()

```